### ISSUE TITLE
List available test names

### ISSUE THREAD

**Issue by dtrv** [bats-core/bats-core#701]
**Is your feature request related to a problem? Please describe.**
Running `bats -rc .` or `bats -c <somedir>` prevents for starting to much tests at once. Great.
After that I filter the test I'd like to run with `-f <pattern>`. But one have to lookup for the 'pattern' first.

**Describe the solution you'd like**
`-l, --list      List test cases names without running any tests`
This should be compatible with `--recursive` and `--filter` 

**Describe alternatives you've considered**
`grep -r '^@test' *bats | grep <pattern>` ... but it's not nice

**Comment by martin-schulze-vireso** [comment]
Did you see #508?

**Comment by dtrv** [comment]
No, I didn't. Thanks. But the suggestes solution `BATS_RUN_TMPDIR=your/target/dir bats --no-tempdir-cleanup --count test.bats` is IMHO as unattractive as my suggestion to use grep.
If there is an argument for `--count` and for `--filter` it would be nice to have a `--list` as well.

**Comment by cniethammer** [comment]
Hello,

I was looking for a `--list` or `--dryrun` option and came across this issue. Are there any plans or activities implementing this since this issue was opened?
